docs/development/design/stack.md

```diff
@@ -0,0 +1,60 @@
+# Nexodus Stack
+
+## Summary
+
+This proposal documents the decisions made around the Nexodus software stack.
+
+## Proposal
+
+### Kubernetes Only
+
+#### Background
+
+Early on, we used Docker Compose for orchestration.
+As the stack and complexity grew, we decided to exclusively support Kubernetes, requiring local development to use KIND.
+
+To support use of KIND for local development, and OpenShift for production deployment we settled on using Kustomize.
+
+#### Decisions
+
+- We only support deployment on K8s
+- Kustomize is our deployment configuration tool
+- All manifests in `deploy/base` MUST be useable in any K8s distribution.
+- We allow operators to be used, provided they may also be installed on any K8s distribution.
+
+### Microservices First
```

Comment:
I think some of this could go into `docs/development/development.md` in the "guiding principles" section. The longer text in here is good, but I'd like to come up with a 1 or 2 sentence summary that captures the intent of it.  Here's a first try:

- **Modularity** - We embrace microservices and modularity in general, though we will compromise if being non-modular is simpler or offers a better developer or user experience. See `[Nexodus Stack](stack.md)` for additional comments on this.